[Caffe2] Build perfkernels targeting SVE128

Nicoshev opened the pull request:
Summary: We are now building perfkernels using SVE/Neon enhancements

Test Plan:
Sigrid Predictor canary

Rollback Plan:

Differential Revision: D78902495


cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

Skylion007 commented on aten/src/ATen/cpu/vec/sve/vec_common_sve.h:
```diff
@@ -27,6 +27,33 @@ namespace at::vec {
+}
+template <typename T>
+std::ostream& operator<<(std::ostream& stream, const Vectorized<T>& vec) {
+  T buf[Vectorized<T>::size()];
```
Yikes dynamic array... is size constexpr? Use std::array if so

Skylion007 commented on aten/src/ATen/cpu/vec/sve/vec_common_sve.h:
```diff
@@ -27,6 +27,33 @@ namespace at::vec {
+std::ostream& operator<<(std::ostream& stream, const Vectorized<T>& vec) {
+  T buf[Vectorized<T>::size()];
+  vec.store(buf);
+  stream << "vec[";
```
This can be cleaned up a lot with fmtlib and fmt::streamed and fmt::join

Skylion007 commented on aten/src/ATen/cpu/vec/functional_base.h:
```diff
@@ -102,31 +102,59 @@ struct VecReduceAllSIMD<float, Op> {
-    Vec v1 = v1_1;
+    float32x4_t v1_1 = vextq_f32(vReg, vReg, 2);
+
+     __at_align__ float v1[4];
```
Can you not do this with std::array and data?

Commit:
[Caffe2] Import SVE128 PR (#158932)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/158932

Importing https://github.com/pytorch/pytorch/pull/138388, as it improves SVE support for perfkernels

Test Plan: We will test it on AdFinder/AdRetriever/AdRanker offline tier

Differential Revision: D70788867

Reviewed By: r1mikey

Commit:
[Caffe2] Enable SVE128

Summary: Enabling compilation targeting SVE128

Test Plan: AdRanker/AdFinder ServiceLab

Differential Revision: D78691521

Nicoshev committed:
[Caffe2] Differentiate SVE128 from SVE for vectorization (#159274)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/159274

We are introducing the SVE128 vectorized<> layer.

Idea is to differentiate SVE128 perfkernels from the general SVE implementation

Mixing NEON and SVE should maximize performance on SVE128 cpus

Test Plan:
Sigrid Predictor canary

Rollback Plan:

Differential Revision: D78902495
